modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/Semialign.kt

```diff
@@ -55,4 +58,40 @@ interface Semialign<F> : Functor<F> {
    * ```
    */
   fun <A, B, C> alignWith(a: Kind<F, A>, b: Kind<F, B>, fa: (Ior<A, B>) -> C): Kind<F, C> = align(a, b).map(fa)
+
+  /**
+   * aligns two structures and combine them with the given Semigroups '+'
+   */
+  fun <A> Kind<F, A>.salign(
+    SG: Semigroup<A>,
+    other: Kind<F, A>
+  ): Kind<F, A> =
+    alignWith(this, other) {
+      it.fold(::identity, ::identity) { a, b ->
+        SG.run { a.combine(b) }
+      }
+    }
+
+  /**
+   * Align two structures as in zip, but filling in blanks with None.
+   */
+  fun <A, B> Kind<F, A>.padZip(
+    other: Kind<F, B>
+  ): Kind<F, Tuple2<Option<A>, Option<B>>> =
+    alignWith(this, other) { ior ->
+      when (val e = ior.bimap({ Option.just(it) }, { Option.just(it) })) {
+        is Ior.Right -> Option.empty<A>() toT e.value
+        is Ior.Left -> e.value toT Option.empty()
+        is Ior.Both -> e.leftValue toT e.rightValue
```

Comment:
This would be cleaner with `Ior.fold`.

Reply:
I think this is still open and after that this pr is good to merge

Reply:
true, i missed that ...